File: ktmall/ktmall_server/app/models/category.go
package models

const (
	CategoryTableName = "category"
)

// 商品分类
type Category struct {
	BaseModel
	Name     string
	Icon     string
	ParentId uint
}

type CategorySerializer struct {
	ID         uint   `json:"id"`
	Name       string `json:"name"`
	Icon       string `json:"icon"`
	ParentId   uint   `json:"parentId"`
	IsSelected bool   `json:"isSelected"`
}

func (Category) TableName() string {
	return CategoryTableName
}

func (m *Category) Serialize() CategorySerializer {
	return CategorySerializer{
		ID:         m.ID,
		Name:       m.Name,
		Icon:       m.Icon,
		ParentId:   m.ParentId,
		IsSelected: false,
	}
}

File: ktmall/ktmall_app/lib_goods_center/src/main/java/net/away0x/lib_goods_center/data/protocol/Category.kt
package net.away0x.lib_goods_center.data.protocol

/*
     商品分类
 */
data class Category(
        val id: Int, //分类ID
        val name: String, //分类名称
        val icon: String = "", //分类图标
        val parentId: Int, //分类 父级ID
        var isSelected: Boolean = false//是否被选中
)
